From ed16690bf57d80eb980fa10999ef7536e328d101 Mon Sep 17 00:00:00 2001
From: Shengzhou Liu <Shengzhou.Liu@freescale.com>
Date: Thu, 15 Nov 2012 19:35:51 +0000
Subject: [PATCH 149/227] dts/FMANv3: update dtsi for MDIO support in FMANv3

- add the two dedicated MDIO nodes in FMAN0 and FMAN1 for external PHYs.
- correct MDIO nodes of each MEMAC for tbi.

Signed-off-by: Shengzhou Liu <Shengzhou.Liu@freescale.com>
Signed-off-by: Andy Fleming <afleming@freescale.com>
[Kevin: Original patch taken from fsl sdk 1.3.1
QorIQ-SDK-V1.3.1-SOURCE-20121220-yocto.iso.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-0.dtsi |    8 +++-----
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-1.dtsi |    8 +++-----
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-0.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-1.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-2.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-3.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-4.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-5.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-0.dtsi       |   14 ++++++++++++++
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-0.dtsi |   14 ++++++--------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-1.dtsi |   14 ++++++--------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-0.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-1.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-2.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-3.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-4.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-5.dtsi  |   10 ++++------
 arch/powerpc/boot/dts/fsl/qoriq-fman3-1.dtsi       |   14 ++++++++++++++
 18 files changed, 94 insertions(+), 98 deletions(-)

diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-0.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-0.dtsi
index ee59ee4..72c306c 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-0.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-0.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		reg = <0xf0000 0x1000>;
 		fsl,port-handles = <&fman0_10g_rx0 &fman0_10g_tx0>;
 	};
-/*
-	xmdio0: mdio@f1000 {
+
+	mdio@f1000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-xmdio";
+		compatible = "fsl,fman-memac-tbi";
 		reg = <0xf1000 0x1000>;
-		interrupts = <100 1 0 0>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-1.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-1.dtsi
index fb5f7d9..c53dadc 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-1.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-10g-1.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		reg = <0xf2000 0x1000>;
 		fsl,port-handles = <&fman0_10g_rx1 &fman0_10g_tx1>;
 	};
-/*
-	xmdio0: mdio@f3000 {
+
+	mdio@f3000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-xmdio";
+		compatible = "fsl,fman-memac-tbi";
 		reg = <0xf3000 0x1000>;
-		interrupts = <100 1 0 0>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-0.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-0.dtsi
index 9051800..5d34959 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-0.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-0.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		fsl,port-handles = <&fman0_rx0 &fman0_tx0>;
 		ptimer-handle = <&ptp_timer0>;
 	};
-/*
-	mdio0: mdio@e1120 {
+
+	mdio@e1000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-mdio";
-		reg = <0xe1120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe1000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-1.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-1.dtsi
index 8b0d411..39620a3 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-1.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-1.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		fsl,port-handles = <&fman0_rx1 &fman0_tx1>;
 		ptimer-handle = <&ptp_timer0>;
 	};
-/*
-	mdio@e3120 {
+
+	mdio@e3000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe3120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe3000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-2.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-2.dtsi
index 0659794..9c1fb1a 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-2.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-2.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		fsl,port-handles = <&fman0_rx2 &fman0_tx2>;
 		ptimer-handle = <&ptp_timer0>;
 	};
-/*
-	mdio@e5120 {
+
+	mdio@e5000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe5120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe5000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-3.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-3.dtsi
index 18306d6..5d2ba1a 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-3.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-3.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		fsl,port-handles = <&fman0_rx3 &fman0_tx3>;
 		ptimer-handle = <&ptp_timer0>;
 	};
-/*
-	mdio@e7120 {
+
+	mdio@e7000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe7120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe7000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-4.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-4.dtsi
index cd051ef..4b1820b 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-4.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-4.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		fsl,port-handles = <&fman0_rx4 &fman0_tx4>;
 		ptimer-handle = <&ptp_timer0>;
 	};
-/*
-	mdio@e9120 {
+
+	mdio@e9000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe9120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe9000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-5.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-5.dtsi
index b2728ca..aa06d13 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-5.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0-1g-5.dtsi
@@ -52,13 +52,11 @@ fman@400000 {
 		fsl,port-handles = <&fman0_rx5 &fman0_tx5>;
 		ptimer-handle = <&ptp_timer0>;
 	};
-/*
-	mdio@eb120 {
+
+	mdio@eb000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xeb120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xeb000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0.dtsi
index 9fc9320..0e1e79c 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-0.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-0.dtsi
@@ -124,6 +124,20 @@ fman0: fman@400000 {
 		reg = <0xc7000 0x1000>;
 	};
 
+        mdio@fc000 {
+                #address-cells = <1>;
+                #size-cells = <0>;
+                compatible = "fsl,fman-memac-mdio";
+                reg = <0xfc000 0x1000>;
+        };
+
+        mdio@fd000 {
+                #address-cells = <1>;
+                #size-cells = <0>;
+                compatible = "fsl,fman-memac-mdio";
+                reg = <0xfd000 0x1000>;
+        };
+
 	ptp_timer0: rtc@fe000 {
 		compatible = "fsl,fman-rtc";
 		reg = <0xfe000 0x1000>;
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-0.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-0.dtsi
index 39e1348..b63fb54 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-0.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-0.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		reg = <0xf0000 0x1000>;
 		fsl,port-handles = <&fman1_10g_rx0 &fman1_10g_tx0>;
 	};
-/*
-	xmdio0: mdio@f1000 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "fsl,fman-xmdio";
-		reg = <0xf1000 0x1000>;
-		interrupts = <100 1 0 0>;
+
+	mdio@f1000 {
+                #address-cells = <1>;
+                #size-cells = <0>;
+		compatible = "fsl,fman-memac-tbi";
+                reg = <0xf1000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-1.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-1.dtsi
index 92b8f96..56cb7b1 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-1.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-10g-1.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		reg = <0xf2000 0x1000>;
 		fsl,port-handles = <&fman1_10g_rx1 &fman1_10g_tx1>;
 	};
-/*
-	xmdio0: mdio@f3000 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "fsl,fman-xmdio";
-		reg = <0xf3000 0x1000>;
-		interrupts = <100 1 0 0>;
+
+	mdio@f3000 {
+                #address-cells = <1>;
+                #size-cells = <0>;
+		compatible = "fsl,fman-memac-tbi";
+                reg = <0xf3000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-0.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-0.dtsi
index d809af6..6a4fea5 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-0.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-0.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		fsl,port-handles = <&fman1_rx0 &fman1_tx0>;
 		ptimer-handle = <&ptp_timer1>;
 	};
-/*
-	mdio@e1120 {
+
+	mdio@e1000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe1120 0xee0>;
-		interrupts = <100 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe1000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-1.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-1.dtsi
index 40e2720..80f0cd9 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-1.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-1.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		fsl,port-handles = <&fman1_rx1 &fman1_tx1>;
 		ptimer-handle = <&ptp_timer1>;
 	};
-/*
-	mdio@e3120 {
+
+	mdio@e3000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe3120 0xee0>;
-		interrupts = <101 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe3000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-2.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-2.dtsi
index 752e70c..a0cbf96 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-2.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-2.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		fsl,port-handles = <&fman1_rx2 &fman1_tx2>;
 		ptimer-handle = <&ptp_timer1>;
 	};
-/*
-	mdio@e5120 {
+
+	mdio@e5000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe5120 0xee0>;
-		interrupts = <101 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe5000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-3.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-3.dtsi
index 7f31ab0..636ff3e 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-3.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-3.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		fsl,port-handles = <&fman1_rx3 &fman1_tx3>;
 		ptimer-handle = <&ptp_timer1>;
 	};
-/*
-	mdio@e7120 {
+
+	mdio@e7000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe7120 0xee0>;
-		interrupts = <101 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe7000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-4.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-4.dtsi
index 934ff65..ba12e35 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-4.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-4.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		fsl,port-handles = <&fman1_rx4 &fman1_tx4>;
 		ptimer-handle = <&ptp_timer1>;
 	};
-/*
-	mdio@e9120 {
+
+	mdio@e9000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xe9120 0xee0>;
-		interrupts = <101 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xe9000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-5.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-5.dtsi
index 5fc8034..c8d145e 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-5.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1-1g-5.dtsi
@@ -52,13 +52,11 @@ fman@500000 {
 		fsl,port-handles = <&fman1_rx5 &fman1_tx5>;
 		ptimer-handle = <&ptp_timer1>;
 	};
-/*
-	mdio@eb120 {
+
+	mdio@eb000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,fman-tbi";
-		reg = <0xeb120 0xee0>;
-		interrupts = <101 1 0 0>;
+		compatible = "fsl,fman-memac-tbi";
+		reg = <0xeb000 0x1000>;
 	};
-*/
 };
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1.dtsi
index 35e5637..35b09c8 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman3-1.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman3-1.dtsi
@@ -124,6 +124,20 @@ fman1: fman@500000 {
 		reg = <0xc7000 0x1000>;
 	};
 
+	mdio@fc000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "fsl,fman-memac-mdio";
+		reg = <0xfc000 0x1000>;
+	};
+
+	mdio@fd000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "fsl,fman-memac-mdio";
+		reg = <0xfd000 0x1000>;
+	};
+
 	ptp_timer1: rtc@fe000 {
 		compatible = "fsl,fman-rtc";
 		reg = <0xfe000 0x1000>;
-- 
1.7.9.7

